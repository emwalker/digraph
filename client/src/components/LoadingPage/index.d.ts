import React from 'react';
import { LocationType } from 'components/types';
type Props = {
    location: LocationType;
};
declare const LoadingPage: ({ location }: Props) => React.JSX.Element;
export default LoadingPage;
