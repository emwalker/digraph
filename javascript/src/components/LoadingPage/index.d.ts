import { Router } from 'found';
import { LocationType } from 'components/types';
declare type Props = {
    location: LocationType;
    router: Router;
};
declare const LoadingPage: ({ location, router }: Props) => JSX.Element;
export default LoadingPage;
