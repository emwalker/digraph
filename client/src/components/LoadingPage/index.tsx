import React from 'react'

import { LocationType } from 'components/types'
import Page from 'components/ui/Page'
import SearchBoxPlaceholder from 'components/ui/SearchBox/Placeholder'
import Columns from 'components/ui/Columns'
import LeftColumn from 'components/ui/LeftColumn'
import RightColumn from 'components/ui/RightColumn'
import SidebarList from 'components/ui/SidebarList'

type Props = {
  location: LocationType,
}

const LoadingPage = ({ location }: Props) => {
  const state = location.state || {}
  const { itemTitle } = state

  return (
    <Page>
      <div className="Subhead clearfix gutter">
        <div className="Subhead-heading col-lg-8 col-12">
          { itemTitle }
        </div>
        <SearchBoxPlaceholder />
      </div>
      <div>
        <Columns>
          <RightColumn>
            <SidebarList
              title="Parent topics"
              placeholder="There are no parent topics for this topic."
              items={[]}
            />
          </RightColumn>
          <LeftColumn>
            <div className="blankslate">
              <p>Searching the computers for items ...</p>
            </div>
          </LeftColumn>
        </Columns>
      </div>
    </Page>
  )
}

export default LoadingPage
