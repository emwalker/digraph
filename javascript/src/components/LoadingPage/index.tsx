import React from 'react'
import { GoRepo } from 'react-icons/go'
import { Router } from 'found'

import { LocationType } from 'components/types'
import Page from 'components/ui/Page'
import SearchBox from 'components/ui/SearchBox'
import Columns from 'components/ui/Columns'
import LeftColumn from 'components/ui/LeftColumn'
import RightColumn from 'components/ui/RightColumn'
import SidebarList from 'components/ui/SidebarList'

type Props = {
  location: LocationType,
  router: Router,
}

const LoadingPage = ({ location, router }: Props) => {
  const state = location.state || {}
  const { orgLogin, repoName, itemTitle } = state

  return (
    <Page>
      <nav aria-label="Breadcrumb" className="mb-1">
        <ol>
          <li className="breadcrumb-item">
            <GoRepo className="mr-1" />
            {' '}
            <a href="#">{orgLogin}</a>
          </li>
          <li
            className="breadcrumb-item breadcrumb-item-selected text-gray"
            aria-current="page"
          >
            { repoName }
          </li>
        </ol>
      </nav>
      <div className="Subhead clearfix gutter">
        <div className="Subhead-heading col-lg-8 col-12">
          { itemTitle }
        </div>
        <SearchBox
          className="col-lg-4 col-12"
          location={location}
          router={router}
          value=""
        />
      </div>
      <div>
        <Columns>
          <RightColumn>
            <SidebarList
              title="Parent topics"
              orgLogin={orgLogin}
              placeholder="There are no parent topics for this topic."
              repoName={repoName || 'unknown'}
              items={[]}
            />
          </RightColumn>
          <LeftColumn>
            <div className="blankslate">
              <p>Searching the computers for items ...</p>
            </div>
          </LeftColumn>
        </Columns>
      </div>
    </Page>
  )
}

export default LoadingPage
