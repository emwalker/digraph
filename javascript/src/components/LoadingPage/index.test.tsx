import React from 'react'
import { shallow } from 'enzyme'

import LoadingPage from './index'

const props = {
  location: {
    pathname: '/some/path',
    state: {
      orgLogin: 'Gnusto',
      repoName: 'General collection',
      itemTitle: 'Some title',
    },
  },
}

describe('<LoadingPage />', () => {
  // @ts-expect-error
  const wrapper = shallow(<LoadingPage {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
