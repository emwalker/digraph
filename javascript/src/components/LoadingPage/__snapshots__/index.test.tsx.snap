// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingPage /> renders 1`] = `
<Page>
  <nav
    aria-label="Breadcrumb"
    className="mb-1"
  >
    <ol>
      <li
        className="breadcrumb-item"
      >
        <GoRepo
          className="mr-1"
        />
         
        <a
          href="#"
        >
          Gnusto
        </a>
      </li>
      <li
        aria-current="page"
        className="breadcrumb-item breadcrumb-item-selected text-gray"
      >
        General collection
      </li>
    </ol>
  </nav>
  <div
    className="Subhead clearfix gutter"
  >
    <div
      className="Subhead-heading col-lg-8 col-12"
    >
      Some title
    </div>
    <Relay(SearchBox)
      className="col-lg-4 col-12"
      location={
        Object {
          "pathname": "/some/path",
          "state": Object {
            "itemTitle": "Some title",
            "orgLogin": "Gnusto",
            "repoName": "General collection",
          },
        }
      }
      value=""
    />
  </div>
  <div>
    <Columns>
      <RightColumn>
        <_default
          items={Array []}
          orgLogin="Gnusto"
          placeholder="There are no parent topics for this topic."
          repoName="General collection"
          title="Parent topics"
        />
      </RightColumn>
      <LeftColumn>
        <div
          className="blankslate"
        >
          <p>
            Searching the computers for items ...
          </p>
        </div>
      </LeftColumn>
    </Columns>
  </div>
</Page>
`;
